### ISSUE TITLE
Batch rename folders to remove _tmp

### ISSUE THREAD

**Issue by iamsunwei** [tachiyomiorg/tachiyomi#6868]
### Describe your suggested feature

This is NOT a duplicate of #1501. The issue and the "fix" presented in #1193 is a band-aid that simply makes the badges ignore the existence of the (_tmp) folder. This does not fix the fact that the (_tmp) will still have working and fully downloaded pages in the folder and is NOT deleted or renamed to be readable. You'll see people erroneously making bug requests about this exact problem and those issues get buried because of the bot. examples : #5318 #4390 #4379 

#5425 @arkon mentions removing the (_tmp) manually, however the issue with that is that occasionally when you download many chapters at once, Tachiyomi will have the same issue many times. This means you need to manually rename upwards of 40+ chapters. I am simply asking for a function in-app to batch rename. This would solve many people's issues with this particular problem where the downloaded chapter is finished (has all pages) but is not renamed by Tachiyomi automatically. The issue in #3778 is apparently fixed via an android security update however this is clearly not the case as I and numerous others are still have the same issue, hence the need for this feature. 

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/tachiyomi-extensions/issues/new/choose).
- [X] I have updated the app to version **[0.13.1](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I will fill out all of the requested information in this form.

**Comment by arkon** [comment]
From https://github.com/tachiyomiorg/tachiyomi/issues/6867#issuecomment-1089383172:

> Batch renaming isn't a valid solution because it wouldn't know if the downloads are actually valid or not. The actual root issue needs to be addressed, which you could create a new issue for if there isn't already one.

Your original issue didn't really provide many details, but are you able to see a pattern to which folders fail to get renamed, or are you able to consistently reproduce it? If we can figure out what cases this is happening (such as the issue that was in the original Android 11 release), then it'd be easier for us to actually address that.

**Comment by iamsunwei** [comment]
Unfortunately, it does seem to be happening entirely at random. There's no discernable pattern that can be seen when looking at folders the haven't been successfully renamed and I've been looking for a pattern for the last year and a half that I've been having this issue. The only thing I've noticed is that occasionally it will fail to rename multiple chapters in a row and then be fine for the next few, fail one, be fine, and then fail to rename multiple again. 

This is pure conjecture, but I think it may have something to do with Tachiyomi failing to rename the folder while it is still downloading the chapter, so before it's completely finished, Tachiyomi may attempt to rename it, and it is still simultaneously downloading to the folder and as such, re-marks it as (_tmp) and then finishes the download. But because perhaps it only attempts the renaming once, it never renames it. Again, I'm unsure if this is the case but I've noticed that because it tends to be many chapters in a row, some slower download speeds or poor connectivity could cause some hiccups in Tachiyomi thinking a chapter is completely finished (some files may be in progress and it has already moved on) 

On the other hand, if there IS a chapter that is getting held up, it WILL show up in the download queue for a bit. I currently have one that is 12/12 pages finished but is stuck in the download queue. Hitting the resume button causes it to re-attempt the download but immediately says "Download Finished". The chapter is not removed from the queue, and if I go into the file explorer, that particular chapter still has its _tmp tag.

**Comment by iamsunwei** [comment]
Upon further inspection of this particular chapter (The weakest class is the blacksmith, but it's actually the strongest is the name of the series), Chapter 12 is marked in Tachiyomi's download queue as having 12 total pages. It shows 12/12 and is stuck, and also still has the residual (_tmp) tag. HOWEVER, I have noticed that _the folder itself has 13 files in it, all valid pages_ so, my working theory is that Tachiyomi detects a smaller number of total pages that need to be downloaded, but actually downloads the correct amount and is having an issue where it sees 13 pages for example, but knows there needs to be exactly 12, and thinks it's not done yet.
![Screenshot_2022-04-05-15-05-15-215](https://user-images.githubusercontent.com/24500241/161857842-e251171b-3bc8-4682-be0c-e0f7bdb500cc.png)
![Screenshot_2022-04-05-15-05-23-458](https://user-images.githubusercontent.com/24500241/161857856-76cc560b-4eb2-44db-ad7d-cc036f405ff1.png)
![Screenshot_2022-04-05-15-05-29-981](https://user-images.githubusercontent.com/24500241/161857863-dc9e8f5a-ef5d-4765-9be5-e66b019ae71c.png)

**Comment by FourTOne5** [comment]
The pattern is that a `.nomedia` file exists in the temp folder in this case the downloader for some reason can't rename the folder. I actually forgot about this big cause it happened time to time and then I moved to cbz downloader

**Comment by iamsunwei** [comment]
> The pattern is that a `.nomedia` file exists in the temp folder in this case the downloader for some reason can't rename the folder. I actually forgot about this big cause it happened time to time and then I moved to cbz downloader

Does the cbz downloader solve this issue? If so this may be a very solid workaround the problem for now that can be recommended instead of a manual renaming

**Comment by FourTOne5** [comment]
> > The pattern is that a `.nomedia` file exists in the temp folder in this case the downloader for some reason can't rename the folder. I actually forgot about this big cause it happened time to time and then I moved to cbz downloader
> 
> Does the cbz downloader solve this issue? If so this may be a very solid workaround the problem for now that can be recommended instead of a manual renaming

Haven't had this issue after using CBZ downloader

**Comment by orangpelupa** [comment]
Can confirm this issue still randomly occur in the latest tachiyomi. 0. 14.5

The Workaround of manually renaming the temporary folder also still work.

**Comment by AlexisTM** [comment]
I have had Tachiyomi for a while and a manual cleanup removed around 8GB of _ChXX and _Chxxx_tmp folders.

**Comment by maxime-rey** [comment]
Hi, also used Tachiyomi for a while and never had any problems up until a few months ago when downloaded chapers did not seem to be downloaded according to the app. Just checked and indeed, the chapter folders just remain as `blabla_tmp. This is systematic for me, and I am on Tachiyomi stable 0.14.7, Android 13, Pixel 4.
I did a test on Kingdom (on MangaSee), erased the whole folder, downloaded a few chapters again, and they indeed stayed as `blabla_tmp`. There was a `.nomedia` file in them. I just ticked the option "save as CBZ archive" and it seems to work fine now. Can read the few I've tested in plane more.

TLDR; Save as CBZ archive seems to fix it.
